src/pyodide/internal/snapshot.ts

```diff
@@ -262,8 +262,13 @@ function memorySnapshotDoImports(Module: Module): string[] {
   const deduplicatedModules = [...new Set(importedModules)];
 
   // Import the modules list so they are included in the snapshot.
-  if (deduplicatedModules.length > 0) {
-    simpleRunPython(Module, 'import ' + deduplicatedModules.join(','));
+  for (const mod of deduplicatedModules) {
+    try {
+      simpleRunPython(Module, `import ${mod}`);
+    } catch (e) {
+      continue;
+    }
+    simpleRunPython(Module, `delete ${mod.split(".", 1)[0]}`);
```

Comment:
Why delete?
And why the split?
I don't understand this we should probably have a comment here if there's a good reason.

Reply:
Delete because we are importing the module for its side effects into a different context than it is actually imported into at runtime. If `a.py` has `import some_mod` here we are importing it into `__main__`. We just want to execute `some_mod` without actually adding symbols.

The reason for the split is if we want to execute submodules. So for instance if `some_mod` has a lazy submodule and we find `import some_mod.submod`, we need to import this dotted path to instantiate the submodule. But `delete some_mod.submod` will remove `submod` from `some_mod`'s namespace which is broken. We want `delete some_mod` to remove `some_mod` from the `__main__` namespace. I can add these as comments.

Reply:
Perhaps it would be simpler to do 
```js
simpleRunPython(Module, `exec("import ${mod}", {})`);
```
and then we wouldn't need to delete it...

Reply:
I'm just confused why this delete suddenly appeared with this change.
Doesn't seem like what you're saying is new or specific to the title of this PR.

Reply:
It should have been there before. See `toDelete` above on line 232.